apollo-router/src/services/supergraph_service.rs

```diff
@@ -1542,4 +1542,543 @@ mod tests {
         context.insert(ACCEPTS_MULTIPART_CONTEXT_KEY, true).unwrap();
         context
     }
+
+    #[tokio::test]
+    async fn interface_object_typename_rewrites() {
+        let schema = r#"
+            schema
+              @link(url: "https://specs.apollo.dev/link/v1.0")
+              @link(url: "https://specs.apollo.dev/join/v0.3", for: EXECUTION)
+            {
+              query: Query
+            }
+
+            directive @join__enumValue(graph: join__Graph!) repeatable on ENUM_VALUE
```

Comment:
It's taking a lot of lines in our file, maybe for later we should refactor this and put all these schemas in separate files in `testdata` directory and use `include_str!` macro to have a cleaner file.

Reply:
Happy to change it if you prefer, it's fairly quick. I mostly followed the pattern of the previous tests in this file.

Though just for the conversion, I tend to find it easier to maintain tests when they are somewhat self-contained, and you don't have to navigate multiple files to understand what is going on. But it's probably more of a personal preference and not trying to push anything really.